relay.go

```diff
@@ -0,0 +1,86 @@
+package sdk
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// The returned RelayRequest struct can be marshalled and delivered to a service endpoint through an HTTP POST request.
+func BuildRelayRequest(
+	endpointSelector EndpointSelector,
+	requestBz []byte,
+) (*servicetypes.RelayRequest, error) {
+	if endpointSelector == nil {
+		return nil, errors.New("BuildRelayRequest: endpointSelector not specified")
+	}
+
+	sessionHeader, err := endpointSelector.SessionHeader()
+	if err != nil {
+		return nil, fmt.Errorf("BuildRelayRequest: could not get session header: %w", err)
+	}
+
+	if err := sessionHeader.ValidateBasic(); err != nil {
+		return nil, fmt.Errorf("BuildRelayRequest: error validating session header: %w", err)
+	}
+
+	supplierAddress, err := endpointSelector.SelectedSupplierAddress()
+	if err != nil {
+		return nil, fmt.Errorf("BuildRelayRequest: error getting a supplier: %w", err)
+	}
+
+	return &servicetypes.RelayRequest{
+		Meta: servicetypes.RelayRequestMetadata{
+			SessionHeader:   sessionHeader,
+			SupplierAddress: supplierAddress,
+		},
+		Payload: requestBz,
+	}, nil
+}
+
+func ValidateRelayResponse(
```

Comment:
It seems that `relayRequest` and `publicKeyFetcher` are passed to `ValidateRelayResponse` just to get the supplier's public key.

It would be better to reduce the arguments to just `ctx`, `relayResponseBz` and `supplierPubKey` and get rid of args that "do not seem" to be directly related to response signature verification.

By having the `supplierPubKey` as an arg. we make it explicit that the consumer needs to get it using the `publicKeyFetcher` that we are currently passing anyway and the supplier address that the consumer would very likely have from previous workflow steps (at worse, he can get it from the `RelayRequest`)

Reply:
Also I'm not against removing this function entirely in favor of a `GetSupplierPubKey` one, where a `supplierAddress` and a `publicKeyFetcher` would be meaningful arguments.

Reply:
Thank you for the review. I have updated the PR to remove the `RelayRequest` input, and explicitly ask for the supplier's address.
On the `PublicKeyFetcher` input argument, your point is definitely valid. However, this is done here to minimize the types of input the function expects (which ultimately forces the user to know about and handle). `VerifySupplierSignature` requires a `cryptotypes.PubKey` instrad of a simple byte slice, and `PublicKeyFetcher` is used here to shield the user from this type (and to remove the external dependency from `relay.go`.